add support for ember template tags

IanVS opened the pull request:
Closes #225.

Adds support for ember's single-file component format (gts + gjs).  

This iterates upon https://github.com/IanVS/prettier-plugin-sort-imports/pull/227, with significant changes.  Now, we will preprocess the source code in a similar way to `prettier-plugin-ember-template-tag`, replacing `<template>` tags (using https://github.com/embroider-build/content-tag) with placeholder comments of the correct length.  Then we'll parse that code, extract and sort import nodes, and substitute the new nodes into the _original_ code string. 

~This does end up adding `content-tag` for all users, but it's a very small utility library so it shouldn't hurt anything.~ It's now an optional peer dependency, and must be installed manually for users of Ember template tags.

RobbieTheWagner commented:
I'm curious why there is a fork though? Seems like if there were improvements, they could be contributed upstream?

IanVS commented:
@RobbieTheWagner did you read my reply to your issue? https://github.com/IanVS/prettier-plugin-sort-imports/issues/228.  I  answered that question there.

IanVS commented:
@fbartho thanks for the review!  

Part of the issue here is that I don't really trust peer dependencies.  I've been bitten in the butt by them on several other projects, and they just turn into a maintenance nightmare.  But, after you raised your concern, I did try testing it out here.  I created a new ember project, added the ember template tag prettier plugin and this plugin, used `pnpm patch` to bring in the changes from this PR (after modifying it to make `content-tag` a peer-dep), created a couple of simple ember components with template tags, and tried to run prettier on them.  I got parse errors because ember itself includes an older version of `content-tag`, and that's what was being chosen unless I explicitly installed `content-tag` with the correct version as a direct dependency of the project.  But pnpm didn't warn me that I needed to do this, I think because I had marked it as an optional peer dependency.  

But I agree that when our package is only ~6mb currently, adding a 2.4mb dependency is quite a hit, especially since only a tiny fraction of users will need it.  So, I've made it a peer dependency and added a runtime check that the version we're seeing is compatible with what we need.

As for your other comment, I've never considered the preprocessor itself as part of our public API. It's not documented anywhere, and I can't think of any reason anyone would need to use it directly.  The next time we make a major release, we can add a `exports` field to our package.json to ensure that nobody is relying on any internal parts of the plugin, but for now I think we're safe to release this in a minor.  Thanks for bringing that up, though!

fbartho reviewed (commented):
I see you’ve configured `content-tag` to be optional, but I haven’t tested yet to see if the base library works without `content-tag` installed.

I’m not sure I’ll have time to test that until Sunday, but if that’s something you’re confident is working, then I have no remaining concerns about this PR.

If you’re willing to wait till then, then I’m happy to try to verify it by then.

IanVS commented:
@fbartho Just a note:  I just learned about `import defer`.  Seems like it could be useful for cases like this where we don't want to try to import a dependency unless some condition is met.  https://devblogs.microsoft.com/typescript/announcing-typescript-5-9/#support-for-import-defer

fbartho commented:
Today I Learned! -- Also I equally got disappointed that it can't be used to avoid importing an optional resource.

RobbieTheWagner commented:
This does not work in my project. I am getting errors like this:

```
app/templates/welcome/auto-start.gts
[error] app/templates/welcome/auto-start.gts: TypeError: Cannot read properties of undefined (reading 'ember-template-tag')
[error]     at get ember-template-tag [as ember-template-tag] (/Users/rwwagner90/shipshape/swach/node_modules/@ianvs/prettier-plugin-sort-imports/lib/src/index.js:72:35)
[error]     at initParser (file:///Users/rwwagner90/shipshape/swach/node_modules/prettier/index.mjs:19898:52)
[error]     at normalizeFormatOptions (file:///Users/rwwagner90/shipshape/swach/node_modules/prettier/index.mjs:19947:24)
[error]     at formatWithCursor (file:///Users/rwwagner90/shipshape/swach/node_modules/prettier/index.mjs:20727:11)
[error]     at Module.<anonymous> (file:///Users/rwwagner90/shipshape/swach/node_modules/prettier/index.mjs:22177:12)
[error]     at async formatFiles (file:///Users/rwwagner90/shipshape/swach/node_modules/prettier/internal/cli.mjs:4213:18)
[error]     at async main (file:///Users/rwwagner90/shipshape/swach/node_modules/prettier/internal/cli.mjs:4865:5)
[error]     at async Module.run (file:///Users/rwwagner90/shipshape/swach/node_modules/prettier/internal/cli.mjs:4811:5)
```

Seems like it cannot find `prettier-plugin-ember-template-tag`? I do have it installed though.
